### DISCUSSION TITLE
Assigning type to the result of a select query

### DISCUSSION THREAD

**Discussion by talonx** [prisma/prisma#23629] [Q&A] [answered]
### Question

I'm trying to use the result of a select query in a create - is there a way to use the result of the query as a generated type and then pass it to the `create`? Prisma newbie here so my approach might be totally wrong.

Model 
```prisma
model UserChannel {
  id               Int             @id @default(autoincrement())
  name             String
  type             String
  userid           String
  data             String
  desc             String
  // user             User            @relation(fields: [userid], references: [id])
  ProviderConfig   ProviderConfig? @relation(fields: [providerConfigId], references: [id])
  providerConfigId Int?
}

model ProviderConfig {
  id        Int           @id @default(autoincrement())
  userid    String        @unique
  // user      User          @relation(fields: [userid], references: [id])
  providers String
  channels  UserChannel[]
}
```
This returns the channels (for a given userid, I've omitted that)
```js
    const channels = await prismadb.userChannel.findMany(
        {
            where: {
                userid: userId
            }
        }
    );
```
I want to use the channels result in a create as below
```js
        const provider = await prismadb.providerConfig.create({
            data: {
                channels: userChannels,
                providers: providerList,
                userid: userId,
            },
        });
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
Added in the question details above.
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Ubuntu 22.04
- Database: MySQL
- Node.js version: 20.6.1
- Prisma 5.11.0

**Comment by ludralph** [comment, marked as answer]
Hi @talonx 👋 

 it appears you want to create a new `ProviderConfig` and associate it with existing `UserChannel` records. In Prisma Client, you can use the `connect` keyword to associate existing records with a new one. However, the `connect` keyword requires the unique identifier(s) of the record(s) you want to connect.
In your case, you can modify your create method as follows:
```js
const provider = await prismadb.providerConfig.create({
    data: {
        channels: {
            connect: channels.map(channel => ({ id: channel.id }))
        },
        providers: providerList,
        userid: userId,
    },
});
```
This code assumes that channels is an array of `UserChannel` records, each with a unique`id`. The map function transforms this array into the format expected by Prisma's Client connect keyword.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by talonx** [reply]
This works. I got as far as the connect keyword from the docs on my own but I was not able to figure out how to get the ids from the "channels".  Thank you!

**Reply by ludralph** [reply]
You're welcome
